mmv1/products/alloydb/User.yaml

```diff
@@ -0,0 +1,112 @@
+  guides:
+    'AlloyDB': 'https://cloud.google.com/alloydb/docs/'
+  api: 'https://cloud.google.com/alloydb/docs/reference/rest/v1/projects.locations.clusters.users/create'
+import_format: ['projects/{{project}}/locations/{{location}}/clusters/{{cluster}}/users/{{user_id}}']
+skip_sweeper: true
+autogen_async: true
+custom_code: !ruby/object:Provider::Terraform::CustomCode
+  custom_import: templates/terraform/custom_import/alloydb_user.go.erb
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: 'alloydb_user_builtin'
+    primary_resource_id: 'user1'
+    vars:
+      alloydb_cluster_name: 'alloydb-cluster'
+      alloydb_cluster_pass: 'cluster_secret'
+      alloydb_cluster_resource_id: 'default'
+      alloydb_instance_name: 'alloydb-instance'
+      alloydb_instance_resource_id: 'default'
+      alloydb_user_name: 'user1'
+      alloydb_user_pass: 'user_secret'
+      network_name: 'alloydb-network'
+    test_vars_overrides:
+      alloydb_cluster_name: '"tf-test-user-cluster-" + acctest.RandString(t, 10)'
+      alloydb_cluster_pass: '"cluster_secret1"'
+      alloydb_cluster_resource_id: '"test-cluster1"'
+      alloydb_instance_name: '"tf-test-instance1"'
+      alloydb_instance_resource_id: '"test-instance1"'
+      alloydb_user_name: '"user_1"'
+      alloydb_user_pass: '"user_1_pass"'
+      network_name: 'acctest.BootstrapSharedTestNetwork(t, "alloydb-user-builtin")'
+    ignore_read_extra:
+      - 'password'
+  - !ruby/object:Provider::Terraform::Examples
+    name: 'alloydb_user_iam'
+    primary_resource_id: 'user2'
+    vars:
+      alloydb_cluster_name: 'alloydb-cluster'
+      alloydb_cluster_resource_id: 'default'
+      alloydb_instance_name: 'alloydb-instance'
+      alloydb_instance_resource_id: 'default'
+      alloydb_cluster_pass: 'cluster_secret'
+      alloydb_user_name: 'user2@foo.com'
+      network_name: 'alloydb-network'
+    test_vars_overrides:
+      alloydb_cluster_name: '"tf-test-user-cluster-" + acctest.RandString(t, 10)'
+      alloydb_cluster_pass: '"cluster_secret2"'
+      alloydb_cluster_resource_id: '"test-cluster2"'
+      alloydb_instance_name: '"tf-test-instance2"'
+      alloydb_instance_resource_id: '"test-instance2"'
+      alloydb_user_name: '"user2@foo.com"'
+      network_name: 'acctest.BootstrapSharedTestNetwork(t, "alloydb-user-iam")'
+parameters:
+  - !ruby/object:Api::Type::ResourceRef
+    name: 'cluster'
+    description: |
+      Identifies the alloydb cluster. Must be in the format
+      'projects/{project}/locations/{location}/clusters/{cluster_id}'
+    required: true
+    immutable: true
+    resource: 'Cluster'
+    imports: 'name'
+    url_param_only: true
+  - !ruby/object:Api::Type::String
+    name: 'userId'
+    required: true
+    immutable: true
+    url_param_only: true
+    description: |
+      The database role name of the user.
+  - !ruby/object:Api::Type::Enum
+    name: 'userType'
+    required: true
```

Comment:
Is this field required. Looks like it's optional in https://cloud.google.com/alloydb/docs/reference/rest/v1/projects.locations.clusters.users. We can still set this as required in the Terraform though. Just want to confirm.

Reply:
If not set, the Create user requests will fail. So just wanted to reflect that in this terraform integration.